frontend/src/scenes/pipeline/PipelineNodeNew.tsx

```diff
@@ -80,25 +98,31 @@ export function PipelineNodeNew(
         return <>Plugin ID {pluginId}</>
     }
     if (batchExportDestination) {
+        if (stage !== PipelineStage.Destination) {
+            return <NotFound object={batchExportDestination} />
+        }
         return <>Batch Export Destination {batchExportDestination}</>
     }
 
+    let targets: TableEntry[] = []
+    let loadingAll = false
     if (stage === PipelineStage.Transformation) {
         // Show a list of transformations
         const { plugins, loading } = useValues(pipelineTransformationsLogic)
-        const targets = Object.values(plugins).map(convertPluginToTableEntry)
-        return nodeOptionsTable(stage, targets, loading)
+        loadingAll = loading
+        targets = Object.values(plugins).map(convertPluginToTableEntry)
     } else if (stage === PipelineStage.Destination) {
         const { plugins, loading } = useValues(pipelineDestinationsLogic)
-        // Show a list of destinations - TODO: add batch export destinations too
-        const targets = Object.values(plugins).map(convertPluginToTableEntry)
-        return nodeOptionsTable(stage, targets, loading)
+        loadingAll = loading
+        const pluginTargets = Object.values(plugins).map(convertPluginToTableEntry)
+        const batchExportTargets = Object.values(batchExportServiceNames).map(convertBatchExportToTableEntry)
+        targets = [...batchExportTargets, ...pluginTargets]
     } else if (stage === PipelineStage.SiteApp) {
         const { plugins, loading } = useValues(frontendAppsLogic)
-        const targets = Object.values(plugins).map(convertPluginToTableEntry)
-        return nodeOptionsTable(stage, targets, loading)
+        targets = Object.values(plugins).map(convertPluginToTableEntry)
+        loadingAll = loading
     }
-    return <>Creation is unavailable for {stage}</>
+    return nodeOptionsTable(stage, targets, loadingAll)
 }
 
 function nodeOptionsTable(stage: PipelineStage, targets: TableEntry[], loading: boolean): JSX.Element {
```

Comment:
MIssed this before, but it should be a JSX component (i.e. capitalized and with `props`)